### ISSUE TITLE
Rootless docker images build not work

### ISSUE THREAD

**Issue by phuongnq** [craftercms/craftercms#7291]
### CI Number

5330

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [x] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

Building images with `rootless` option does not work

### Steps to reproduce

Steps:

1. Build all images with the command:
```
./gradlew buildAllImages -ProotlessDockerImages=true
```
2. Run docker with docker-compose
3. Notice the error and the tomcat & deployer do not start:

```
deployer-1  | ./docker-entrypoint.sh: line 22: /opt/crafter/bin/crafter-deployer/deployer.sh run: No such file or directory
tomcat-1    | ./docker-entrypoint.sh: line 22: /opt/crafter/bin/apache-tomcat/bin/catalina.sh run: No such file or directory
...
deployer-1 exited with code 127
tomcat-1 exited with code 127
```

**Comment by phuongnq** [comment]
@sumerjabri @russdanner This seems to happen after the switch off from `gosu`.
I have updated the following:

* Update the entry points to remove the double quotes as the command `exec "$@"` failed. For example, running `exec "ls -la ."` will always be failed with `sh` (the default shell for Alpine). We should instead run `exec ls -la .`. I updated so that in the case of using root, run `su crafter -c "$*"`. Notice the option `"$*"`, this makes sure all parameters are read as a single string.
 
* An optional update. I removed the option `-S` when creating `crafter` user. This option creates a system user without a login shell. By removing this option, we can remove the param `-s /bin/bash` to use the default shell which is `/bin/sh` for Alpine while running `su crafter -c` commands to set git configuration.
